src/main/java/sirius/web/mails/SMTPConfiguration.java

```diff
@@ -63,13 +63,25 @@ public SMTPConfiguration() {
      * @param settings the SMTP settings
      */
     public SMTPConfiguration(Settings settings) {
-        this(settings.get("mail.host").asString(smtpHost),
-             settings.get("mail.port").asString(smtpPort),
-             settings.get("mail.user").asString(smtpUser),
-             settings.get("mail.password").asString(smtpPassword),
-             settings.get("mail.sender").asString(smtpSender),
-             settings.get("mail.senderName").asString(smtpSenderName),
-             settings.get("mail.useEnvelopeFrom").asBoolean(smtpUseEnvelopeFrom));
+        if (settings.get("mail.host").isFilled()) {
+            host = settings.get("mail.host").getString();
```

Comment:
if its not important that all these settings depend on filled mail.host, maybe `value.replaceEmptyWith(xxx) `can be used?

Reply:
I would keep it like this because it makes no sense to mix the host/port/user/password settings from different places

Reply:
maybe we should move this code outside of the constructor and have _one instance_ with the default config (or an empty one). Doing this in a constructor semms convoluted.

Reply:
+1